src/main/kotlin/com/hypto/iam/server/apis/LoginApi.kt

```diff
@@ -0,0 +1,30 @@
+package com.hypto.iam.server.apis
+
+import com.google.gson.Gson
+import com.hypto.iam.server.exceptions.InternalException
+import com.hypto.iam.server.plugins.inject
+import com.hypto.iam.server.security.UserPrincipal
+import com.hypto.iam.server.service.TokenService
+import io.ktor.http.ContentType
+import io.ktor.http.HttpStatusCode
+import io.ktor.server.application.call
+import io.ktor.server.auth.principal
+import io.ktor.server.response.respondText
+import io.ktor.server.routing.Route
+import io.ktor.server.routing.post
+
+fun Route.loginApi() {
+    val tokenService: TokenService by inject()
+    val gson: Gson by inject()
+
+    post("/login") {
+        val principal = context.principal<UserPrincipal>()
+            ?: throw InternalException("Unable to validate request. Please check request")
```

Comment:
I think InternalException is translated as 500. Here the status code that we should return is 401 (UserNotAuthorized).